ccan_compat.h

```diff
@@ -1,7 +1,12 @@
 #ifndef LIGHTNING_CCAN_COMPAT_H
 #define LIGHTNING_CCAN_COMPAT_H
+
 /* Magical file included from config.h (ie. everywhere) which renames
  * sha256 routines so they don't clash with libwally-core's internal ones */
+
+/* So, for obvious reasons, this is an exception to the usual rule that we
+#include "config.h"
```

Comment:
Well, include guards protect against circular inclusions...

On the other hand, yeah, an exception for this is totally better.